### ISSUE TITLE
Inconsistent parameter naming and types

### ISSUE THREAD

**Issue by guidedways** [dropbox/dropbox-sdk-obj-c#129]
This bit me a couple of times recently:

```objc
[[[DBClientsManager authorizedClient].filesRoutes uploadUrl: uploadPath
                                                         mode: mode
                                                   autorename: @(NO)
                                               clientModified: nil
                                                         mute: @(YES)
                                                     inputUrl: [NSURL fileURLWithPath: localFileName]]
```

Note: 
`uploadUrl` says 'URL' in the name but accepts a `NSString`
`inputUrl` says 'URL' in the name but accepts a NSURL

These should either both accept `NSString`s or both should accept `NSURL`s. Very confusing.

**Comment by scobbe** [comment]
@guidedways: This is a good suggestion. I will update `inputUrl` to be a `NSString`. Thanks for the feedback!

**Comment by scobbe** [comment]
@guidedways: This is fixed in the latest version of the SDK, [v3.0.11](https://github.com/dropbox/dropbox-sdk-obj-c/releases/tag/3.0.11).
